src/DataModel/IApiDocumentation.ts

```diff
@@ -0,0 +1,9 @@
+import {IMetadataDescription} from "./IMetadataDescription";
+import {IData} from "./IData";
+
+export interface IApiDocumentation extends IMetadataDescription
+{
+    entryPoint: string;
+
+    getEntryPoint(): Promise<IData>;
```

Comment:
Well, not so different. For simple Urls we'll need to call client's method, but for strongly typed resources we will use their methods - it's not really consistent.
As for the implementation hint - I had to be really tired when I was writing it :). It's so obvious this should be client's method called here